components/places/src/bookmark_sync/engine.rs

```diff
@@ -490,14 +491,42 @@ fn apply_remote_items(db: &PlacesDb, scope: &SqlInterruptScope, now: Timestamp)
         }
     ).ok(); // We ignore the error because it's expected, users should not have entries with the same id that have different types
 
+    // Same as above but we check if any users have any undetected orphaned bookmarks and
+    // report them appropiately
+    let orphaned_count: i64 = db.query_row(
+        "WITH RECURSIVE orphans(id) AS (
```

Comment:
I just want to be sure I understand what's happening here. It looks like you're querying for any bookmarks without parents and any dupes of that orphan. Is that correct? If so, LGTM.

Reply:
Yes! Basically just checking if there is a possibility of existing orphans (which should be impossible but yeknow...), since the tree could be deeply nested, the recursive helps ensure it's not happening at any level